lib/orderbook/OrderBook.ts

```diff
@@ -118,26 +119,58 @@ class OrderBook extends EventEmitter {
     return outgoingOrder ;
   }
 
+  /**
+   * Checks that a currency advertised by a peer are known to us, have a swap client identifier,
+   * and that their token identifier matches ours.
+   */
+  private isPeerCurrencySupported = async (peer: Peer, currency: string) => {
+    const currencyAttributes = this.getCurrencyAttributes(currency);
+    if (!currencyAttributes) {
+      return false; // we don't know about this currency
+    }
+
+    if (!peer.getIdentifier(currencyAttributes.swapClient, currency)) {
+      return false; // peer did not provide a swap client identifier for this currency
+    }
+
+    // temporary check for a direct channel in raiden
+    const swapClient = this.swaps.swapClientManager.get(currency)!;
+    if (isRaidenClient(swapClient)) {
+      const channels = await swapClient.getChannels(swapClient.tokenAddresses.get(currency));
+      let doesDirectChannelExist = false;
+      for (const channel of channels) {
+        if (channel.partner_address && channel.partner_address === peer.raidenAddress) {
+          doesDirectChannelExist = true;
+          break;
+        }
+      }
+      if (!doesDirectChannelExist) {
+        return false;
```

Comment:
I don't know if it is just me but this looks more complicated than it should be. Why not move:
```
    // ensure that our token identifiers match
    const ourTokenIdentifier = this.pool.getTokenIdentifier(currency);
    const peerTokenIdentifier = peer.getTokenIdentifier(currency);
    return ourTokenIdentifier === peerTokenIdentifier;
```

Into the the for loop so that it gets executed when a channel is found and if no channel matches this function should just return false. That way there is no need for `doesDirectChannelExist` and the little extra logic to handle it.

Reply:
We'll want to make this check for LND based currencies as well, which won't enter the for loop because it is raiden only. That's why I have the logic above outside the loop. Does that make sense?

Reply:
This is also out of date given the recent change to check for direct channels before each swap.